crates/sui-indexer/src/models_v2/objects.rs

```diff
@@ -293,3 +297,39 @@ impl From<CoinBalance> for Balance {
         }
     }
 }
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+
+    #[test]
+    fn test_canonical_string_of_object_type_for_coin() {
+        let test_obj = Object::new_gas_for_testing();
+        let indexed_obj = IndexedObject::from_object(1, test_obj, None);
+
+        let stored_obj = StoredObject::from(indexed_obj);
+
+        match stored_obj.object_type {
+            Some(t) => {
+                assert_eq!(t, "0x0000000000000000000000000000000000000000000000000000000000000002::coin::Coin<0000000000000000000000000000000000000000000000000000000000000002::sui::SUI>");
+            }
+            None => {
+                panic!("object_type should not be none");
+            }
+        }
```

Comment:
This test seems to suggest that `object_type` is not correct, because the inner type (in the type parameter) did not get canonicalized with a prefix.

P.S. Just assert_eq! that it's `Some("0x0...02::coin:...")` rather than pattern match first, or if that's annoying, `unwrap` on `object_type`.

Reply:
From the implementation, I couldn't tell you why this doesn't work right now... does this test just need updating?

Reply:
(Ah, this is just a typo -- CI picks up that there should indeed be a `0x` on the inner param).

Reply:
Yes sorry I missed updating this one, maybe I should just bring in expect test as well